src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -373,21 +374,35 @@ void StorageObjectStorage::read(
     if (update_configuration_on_read)
         configuration->update(object_storage, local_context);
 
-    if (partition_by && configuration->withPartitionWildcard())
+    auto config_clone = configuration->clone();
```

Comment:
We make a clone every time, but actually change only in specific cases.
May be make here a smart_ptr on original config, and make a clone only when required?